### PR TITLE
HL-29

### PR DESCRIPTION
**Pull request by brettvniekerk**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-29)

- Adds SWR and uses it to fetch real-time data for runners from the api (removes `api.getRunners` as there is no need for it anymore)
- Also fixes commit errors with `format-staged` and `lint` (was getting errors that husky's pre-commit instructions couldn't find the scripts in `package.json`)
- Also removes api context as the api is not stateful - this has been replaced with a single api instance that is imported where the api is needed - this was also done as both the swr fetcher and api need an axios client, and it didnt make sense to define separate axios clients for both

### CODE REVIEW

**Comment by brettvniekerk** [comment]
branch was developed off from `main` before staging branch was introduced - `main` needs to be merged into `development` first before this gets merged

**Review by OptimusOpus — DISMISSED** [review]

File: src/hooks/data/useRunnerData.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { useMemo } from "react";
+import { SignedRunnersResponse } from "../../types";
+import useSwr from "../useSwr";
+
+type Props = {
+  track: string;
+  raceNumber: number;
+};
+
+const useRunnerData = ({ track, raceNumber }: Props) => {
```
Comment (xiangzhe0812) [inline src/hooks/data/useRunnerData.ts:10]: would be good if we can add type for this hook.

File: src/App.tsx [hunk 2]
```diff
@@ -7,13 +6,11 @@ import { WalletModalProvider } from "./providers/WalletModal";
 const App = () => {
   return (
     <GlobalErrorProvider>
-      <ApiProvider>
```
[the remark this answers is not in the document]
Comment (brettvniekerk) [reply src/App.tsx:10]: we don't - i've refactored `Api.ts` to export an instance of the api as it's a static class. using context and hooks was unnecessary as the api did not hold any state

[the remark this answers is not in the document]
Comment (brettvniekerk) [reply src/App.tsx:10]: i've also refactored everywhere `useApi` was to use this new method in this pr

File: src/hooks/data/useRunnerData.ts (hunk 1 above)
Comment (brettvniekerk) [reply src/hooks/data/useRunnerData.ts:10]: image

due to `useSwr` accepting a generic Typescript already infers the return types for everything - so i don't think it's necessary - but happy to add if you think its critical

**Review by OptimusOpus — APPROVED** [review]

**Review by xiangzhe0812 — APPROVED** [review]

**Pull request merged**